src/main/java/saps/archiver/core/DefaultPermanentStorage.java

```diff
@@ -1,15 +1,105 @@
+import java.io.IOException;
+import java.util.Properties;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
 import saps.common.core.model.SapsImage;
+import saps.archiver.interfaces.PermanentStorage;
+import saps.common.core.storage.exceptions.InvalidPropertyException;
+import saps.common.utils.SapsPropertiesConstants;
+
 
 public class DefaultPermanentStorage implements PermanentStorage {
 
-    public boolean archive(SapsImage task) {
-        return false;
-    }
+  public static final Logger LOGGER = Logger.getLogger(DefaultPermanentStorage.class);
+  public static final String NFS_STORAGE_TASK_DIR_PATTERN ="%s" + File.separator + "%s" + File.separator + "%s";
+  public static final String NFS_STORAGE_TASK_URL_PATTERN = "%s" + File.separator + "%s";
+
+  private final String nfsTempStoragePath;
+  private final String nfsPermanentStoragePath;
+  private final String tasksDirName;
+ 
+
+  public DefaultPermanentStorage(Properties properties) throws InvalidPropertyException {
+  
+    this.nfsTempStoragePath = properties.getProperty(SapsPropertiesConstants.SAPS_TEMP_STORAGE_PATH);
+    this.nfsPermanentStoragePath = properties.getProperty(SapsPropertiesConstants.NFS_PERMANENT_STORAGE_PATH);
+    this.tasksDirName = properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR); 
+
+  }
 
-    public boolean delete(SapsImage task) {
-        return false;
+  @Override
+  public String delete(SapsImage task) throws IOException {
+    String taskId = task.getTaskId();
+    String taskDirPath = String.format(NFS_STORAGE_TASK_DIR_PATTERN, nfsPermanentStoragePath, tasksDirName, task.getTaskId());
+    File taskDir = new File(taskDirPath);
+    FileUtils.deleteDirectory(taskDir);
+    
+    return taskId;
+  }
+
+  @Override
+  public boolean archive(SapsImage task) throws IOException {
+      String taskId = task.getTaskId();
+      LOGGER.info("Archiving task [" + taskId + "] to permanent storage.");
+      String nfsTaskDirPath = createTaskDir(tasksDirName, taskId);
+  
+      // Define the list of directories to copy
+      String[] dirsToCopy = {INPUTDOWNLOADING_DIR, PREPROCESSING_DIR, PROCESSING_DIR};
+      
+      // Copy each directory in the list to the NFS task directory
+      for (String dir : dirsToCopy) {
+          String localDir = String.format(SAPS_TASK_STAGE_DIR_PATTERN, nfsTempStoragePath, taskId, dir);
+          copyDirToDir(localDir, nfsTaskDirPath);
+      }
+      return true;
+  }
+  
+
+  private String createTaskDir(String tasksDir, String taskId) throws IOException {
+    File storageDir = new File(nfsPermanentStoragePath);
+    
+    if (!storageDir.exists()) {
+      throw new FileNotFoundException(
+          "The nfs storage directory [" + nfsPermanentStoragePath + "] was not found");
     }
-}
\ No newline at end of file
+    File nfsTaskDir = new File(String.format(NFS_STORAGE_TASK_DIR_PATTERN, nfsPermanentStoragePath, tasksDir, taskId));
+    FileUtils.forceMkdir(nfsTaskDir);
```

Comment:
what if we remove the check in line 71-73? I guess there is a Mkdir function that fail if the parent dir does not exist